### DISCUSSION TITLE
SIM800L EVB

### DISCUSSION THREAD

**Discussion by Erkman84** [micropython/micropython#9931] [Hardware & Peripherals]
Hi there,
I had previously good experience with the pure original SIM800L (which works from LiPo), I've figured out my working codes for esp and RPB Pico.
Rescently I got an SIM800L EVB Module, which works now from 5V (sounds good), I inserted the SIM, added 5V with 2Amp, and the module starts up, after 10second I see, the SIM is registered on network (Led blinking at every 3 seconds). Fine.

Now I would like to communicate with the module. I allways get as answer: b'\x00' tried several commands. Im frustrated now, and do not know what is missing.

gsm = UART(1, baudrate=9600, tx=Pin(4), rx=Pin(5), timeout=2000)
gsm.write('AT+COPS=?\r\n')  # or simply gsm.write('AT\r\n') or whatever you can imagine
print(gsm.read())

--> returns b'\x00'

if I check it with gsm.any() then I get 0.

That's not good.

Wiring (as I mentioned I've had good experience with the original SIM800L), so yes, TX-RX on RX-TX Pins (Rapsberry PICO throws an error if you don't do it the right way), ground (also tried with booth grounds) on the ground.

I do not know what's wrong. Micropython 1.19. so everything is up to date.

Any idea? Thank you.

**Comment by karfas** [comment]
I don't think it will be a difference, but shouldn't this be `gsm.write(b'AT\r\n')` ?
As you mention a 5V module: Are the GPIO/UART voltage levels compatible ? AFAIK, the Pico needs 3.3V on the GPIOs.
Have you **checked** that _something_ comes out of the TX pin of the Pico and the module ? A LED (with or without resistor) is a cheap and handy tool for this.

**Reply by Erkman84** [reply]
Hi, thank you for your fast reply. Oh, yes, gsm.write could be, as it will converted automatically, so for you I've tried with this binary convert but the error is still the same.
The Pico works from USB with my pc, but sure, one ground is connected with the module for uart, so this is not the problem by wiring at this point.

**Reply by Erkman84** [reply]
Finally, I recylced the module, because of possible damage. Other kind of module work fine, so error not on my side.
